libraries/exoplayer/src/main/java/androidx/media3/exoplayer/mediacodec/MediaCodecInfo.java

```diff
@@ -479,6 +479,17 @@ public DecoderReuseEvaluation canReuseCodec(Format oldFormat, Format newFormat)
         }
       }
 
+      // For eac3 and eac3-joc formats, adaptation is possible without reconfiguration or flushing.
+      if (discardReasons == 0 && (MimeTypes.AUDIO_E_AC3_JOC.equals(mimeType)
+          || MimeTypes.AUDIO_E_AC3.equals(mimeType))) {
+        return new DecoderReuseEvaluation(
```

Comment:
Would you be able to add a unit test to the MediaCodecInfoTest file that given a call of canReuseCodec for dolby content of the same mimetype that `REUSE_RESULT_YES_WITHOUT_RECONFIGURATION` is returned.

Reply:
Yes we will add those tests.

Reply:
I force pushed a commit to https://github.com/DolbyLaboratories/media/commit/ade6e4f083b31e1bc5b1fa931f211a6be9940542 which rebased to Google upstream main branch, and added the requested tests.

I'm not sure how to update this pull request with those changes. Please advise if I need to do anything, or whether you can pull that new commit into this pull request (or create a new pull request).

Reply:
I believe that you need to make the commits to your topic branch, `DolbyLaboratories:dlb/ddp-gapless`, and then rebase the topic branch back onto main. With a git --force push command that should update the PR with your requested commits and update the PR's base commit as `main` head.

Without the collaborator access I'm unable to do this for you.

I also made a comment in the referenced commit as well.
https://github.com/DolbyLaboratories/media/commit/ade6e4f083b31e1bc5b1fa931f211a6be9940542#r145885998